balancer_conn_wrappers.go

```diff
@@ -113,7 +113,10 @@ type subConnUpdate struct {
 func (ccb *ccBalancerWrapper) watcher() {
 	for {
 		select {
-		case u := <-ccb.updateCh.Get():
+		case u, ok := <-ccb.updateCh.Get():
+			if !ok {
+				break
+			}
```

Comment:
Do we need to handle `if !ok { ... }` here? 

The `updateCh` is closed only from `handleClose()` which is called only from this function, and when that happens, we return from this function. So, I feel like there is no way we can hit the `if !ok { ... }` condition.

Reply:
I actually did the first implementation based on those assumptions. Then I switched to more verbose handling because it seemed more right (?). I mean, I can read the code and I know there is no silent side effects if a buffer is closed without diving into the implementation itself. Given the fact the buffers are struct's fields closed somewhere else (eg. a func called from the func they're used), I was thinking it was better to handle it explicitly.
After writing this, I'm no longer so sure. What's your final thought on that @easwars?

Reply:
I sort of agree with the sentiment that the verbose approach might be better in the long run. Today I might immediately be able to figure out that some of these code paths are never executed in reality and hence we can the option of making the code more concise. But small changes implementation can lead to those assumptions not holding true any more. So, I'm ok with taking the verbose approach.

Reply:
Thanks. I'm keeping the verbose approach then so we're consistent across the whole codebase.